src/System.Private.CoreLib/shared/System/Numerics/Vector.cs

```diff
@@ -1406,111 +1406,25 @@ public Vector(Span<T> values)
-                    hash = HashCode.Combine(hash, this.register.int16_6.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.int16_7.GetHashCode());
-                    return hash;
-                }
-                else if (typeof(T) == typeof(uint))
-                {
-                    hash = HashCode.Combine(hash, this.register.uint32_0.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.uint32_1.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.uint32_2.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.uint32_3.GetHashCode());
-                    return hash;
-                }
-                else if (typeof(T) == typeof(int))
-                {
-                    hash = HashCode.Combine(hash, this.register.int32_0.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.int32_1.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.int32_2.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.int32_3.GetHashCode());
-                    return hash;
-                }
-                else if (typeof(T) == typeof(ulong))
-                {
-                    hash = HashCode.Combine(hash, this.register.uint64_0.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.uint64_1.GetHashCode());
-                    return hash;
-                }
-                else if (typeof(T) == typeof(long))
-                {
-                    hash = HashCode.Combine(hash, this.register.int64_0.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.int64_1.GetHashCode());
-                    return hash;
-                }
-                else if (typeof(T) == typeof(float))
-                {
-                    hash = HashCode.Combine(hash, this.register.single_0.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.single_1.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.single_2.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.single_3.GetHashCode());
-                    return hash;
-                }
-                else if (typeof(T) == typeof(double))
-                {
-                    hash = HashCode.Combine(hash, this.register.double_0.GetHashCode());
-                    hash = HashCode.Combine(hash, this.register.double_1.GetHashCode());
-                    return hash;
+                if (typeof(T) == typeof(byte) ||
+                    typeof(T) == typeof(sbyte) ||
+                    typeof(T) == typeof(ushort) ||
+                    typeof(T) == typeof(short) ||
+                    typeof(T) == typeof(int) ||
+                    typeof(T) == typeof(uint) ||
+                    typeof(T) == typeof(float))
```

Comment:
float and double needs to stay separate. GetHashCode for floating point values is normalizing negative zero, etc.